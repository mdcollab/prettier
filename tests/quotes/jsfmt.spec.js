run_spec(__dirname);
run_spec(__dirname, { singleQuote: true });
