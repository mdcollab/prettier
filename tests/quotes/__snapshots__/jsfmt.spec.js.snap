// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions.js 1`] = `
"const a = () => \\"Foo bar\\";

function b(object, key) {
  return object['key'];
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = () => \\"Foo bar\\";

function b(object, key) {
  return object[\\"key\\"];
}
"
`;

exports[`functions.js 2`] = `
"const a = () => \\"Foo bar\\";

function b(object, key) {
  return object['key'];
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const a = () => 'Foo bar';

function b(object, key) {
  return object['key'];
}
"
`;

exports[`strings.js 1`] = `
"// Every string will be changed to double quotes, unless we end up with fewer
// escaped quotes by using single quotes. (Vice versa if the \\"singleQuote\\"
// option is true).
//
// Note that even if a string already has the correct enclosing quotes, it is
// still processed in order to remove unnecessarily escaped quotes within it,
// for consistency.

// Simple strings.
\\"abc\\"
'abc'

// Escape.
'\\\\0'

// Emoji.
'üê∂'

// Empty string.
\\"\\"
''

// Single double quote.
\\"\\\\\\"\\"
'\\"'

// Single single quote.
\\"'\\"
'\\\\''

// Unnecessary escapes.
\\"\\\\'\\"
'\\\\\\"'

// One of each.
\\"\\\\\\"'\\"
'\\"\\\\''

// One of each with unnecessary escapes.
\\"\\\\\\"\\\\'\\"
'\\\\\\"\\\\''

// More double quotes than single quotes.
\\"\\\\\\"'\\\\\\"\\"
'\\"\\\\'\\"'

// More single quotes than double quotes.
\\"\\\\\\"''\\"
'\\"\\\\'\\\\''

// Two of each.
\\"\\\\\\"\\\\\\"''\\"
'\\"\\"\\\\'\\\\''

// Single backslash.
'\\\\\\\\'
\\"\\\\\\\\\\"

// Backslases.
\\"\\\\\\"\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\" '\\\\'\\\\\\\\'\\\\\\\\\\\\'\\\\\\\\\\\\\\\\'\\"
'\\\\'\\\\\\\\\\\\'\\\\\\\\\\\\\\\\\\\\' \\"\\\\\\"\\\\\\\\\\"\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\"'

// Somewhat more real-word example.
\\"He's sayin': \\\\\\"How's it goin'?\\\\\\" Don't ask me why.\\"
'He\\\\'s sayin\\\\': \\"How\\\\'s it goin\\\\'?\\" Don\\\\'t ask me why.'

// Somewhat more real-word example 2.
\\"var backslash = \\\\\\"\\\\\\\\\\\\\\", doubleQuote = '\\\\\\"';\\"
'var backslash = \\"\\\\\\\\\\", doubleQuote = \\\\'\\"\\\\';'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Every string will be changed to double quotes, unless we end up with fewer
// escaped quotes by using single quotes. (Vice versa if the \\"singleQuote\\"
// option is true).
//
// Note that even if a string already has the correct enclosing quotes, it is
// still processed in order to remove unnecessarily escaped quotes within it,
// for consistency.

// Simple strings.
\\"abc\\";
\\"abc\\";

// Escape.
\\"\\\\0\\";

// Emoji.
\\"üê∂\\"; // Empty string.
\\"\\";
\\"\\"; // Single double quote.
'\\"';
'\\"'; // Single single quote.
\\"'\\";
\\"'\\"; // Unnecessary escapes.
\\"'\\";
'\\"'; // One of each.
\\"\\\\\\"'\\";
\\"\\\\\\"'\\"; // One of each with unnecessary escapes.
\\"\\\\\\"'\\";
\\"\\\\\\"'\\"; // More double quotes than single quotes.
'\\"\\\\'\\"';
'\\"\\\\'\\"'; // More single quotes than double quotes.
\\"\\\\\\"''\\";
\\"\\\\\\"''\\"; // Two of each.
\\"\\\\\\"\\\\\\"''\\";
\\"\\\\\\"\\\\\\"''\\"; // Single backslash.
\\"\\\\\\\\\\";
\\"\\\\\\\\\\"; // Backslases.
\\"\\\\\\"\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\" ''\\\\\\\\'\\\\\\\\'\\\\\\\\\\\\\\\\'\\";
'\\\\'\\\\\\\\\\\\'\\\\\\\\\\\\\\\\\\\\' \\"\\"\\\\\\\\\\"\\\\\\\\\\"\\\\\\\\\\\\\\\\\\"'; // Somewhat more real-word example.
\\"He's sayin': \\\\\\"How's it goin'?\\\\\\" Don't ask me why.\\";
\\"He's sayin': \\\\\\"How's it goin'?\\\\\\" Don't ask me why.\\"; // Somewhat more real-word example 2.
'var backslash = \\"\\\\\\\\\\", doubleQuote = \\\\'\\"\\\\';';
'var backslash = \\"\\\\\\\\\\", doubleQuote = \\\\'\\"\\\\';';
"
`;

exports[`strings.js 2`] = `
"// Every string will be changed to double quotes, unless we end up with fewer
// escaped quotes by using single quotes. (Vice versa if the \\"singleQuote\\"
// option is true).
//
// Note that even if a string already has the correct enclosing quotes, it is
// still processed in order to remove unnecessarily escaped quotes within it,
// for consistency.

// Simple strings.
\\"abc\\"
'abc'

// Escape.
'\\\\0'

// Emoji.
'üê∂'

// Empty string.
\\"\\"
''

// Single double quote.
\\"\\\\\\"\\"
'\\"'

// Single single quote.
\\"'\\"
'\\\\''

// Unnecessary escapes.
\\"\\\\'\\"
'\\\\\\"'

// One of each.
\\"\\\\\\"'\\"
'\\"\\\\''

// One of each with unnecessary escapes.
\\"\\\\\\"\\\\'\\"
'\\\\\\"\\\\''

// More double quotes than single quotes.
\\"\\\\\\"'\\\\\\"\\"
'\\"\\\\'\\"'

// More single quotes than double quotes.
\\"\\\\\\"''\\"
'\\"\\\\'\\\\''

// Two of each.
\\"\\\\\\"\\\\\\"''\\"
'\\"\\"\\\\'\\\\''

// Single backslash.
'\\\\\\\\'
\\"\\\\\\\\\\"

// Backslases.
\\"\\\\\\"\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\" '\\\\'\\\\\\\\'\\\\\\\\\\\\'\\\\\\\\\\\\\\\\'\\"
'\\\\'\\\\\\\\\\\\'\\\\\\\\\\\\\\\\\\\\' \\"\\\\\\"\\\\\\\\\\"\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\"'

// Somewhat more real-word example.
\\"He's sayin': \\\\\\"How's it goin'?\\\\\\" Don't ask me why.\\"
'He\\\\'s sayin\\\\': \\"How\\\\'s it goin\\\\'?\\" Don\\\\'t ask me why.'

// Somewhat more real-word example 2.
\\"var backslash = \\\\\\"\\\\\\\\\\\\\\", doubleQuote = '\\\\\\"';\\"
'var backslash = \\"\\\\\\\\\\", doubleQuote = \\\\'\\"\\\\';'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Every string will be changed to double quotes, unless we end up with fewer
// escaped quotes by using single quotes. (Vice versa if the \\"singleQuote\\"
// option is true).
//
// Note that even if a string already has the correct enclosing quotes, it is
// still processed in order to remove unnecessarily escaped quotes within it,
// for consistency.

// Simple strings.
'abc';
'abc';

// Escape.
'\\\\0';

// Emoji.
'üê∂'; // Empty string.
'';
''; // Single double quote.
'\\"';
'\\"'; // Single single quote.
\\"'\\";
\\"'\\"; // Unnecessary escapes.
\\"'\\";
'\\"'; // One of each.
'\\"\\\\'';
'\\"\\\\''; // One of each with unnecessary escapes.
'\\"\\\\'';
'\\"\\\\''; // More double quotes than single quotes.
'\\"\\\\'\\"';
'\\"\\\\'\\"'; // More single quotes than double quotes.
\\"\\\\\\"''\\";
\\"\\\\\\"''\\"; // Two of each.
'\\"\\"\\\\'\\\\'';
'\\"\\"\\\\'\\\\''; // Single backslash.
'\\\\\\\\';
'\\\\\\\\'; // Backslases.
\\"\\\\\\"\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\" ''\\\\\\\\'\\\\\\\\'\\\\\\\\\\\\\\\\'\\";
'\\\\'\\\\\\\\\\\\'\\\\\\\\\\\\\\\\\\\\' \\"\\"\\\\\\\\\\"\\\\\\\\\\"\\\\\\\\\\\\\\\\\\"'; // Somewhat more real-word example.
\\"He's sayin': \\\\\\"How's it goin'?\\\\\\" Don't ask me why.\\";
\\"He's sayin': \\\\\\"How's it goin'?\\\\\\" Don't ask me why.\\"; // Somewhat more real-word example 2.
'var backslash = \\"\\\\\\\\\\", doubleQuote = \\\\'\\"\\\\';';
'var backslash = \\"\\\\\\\\\\", doubleQuote = \\\\'\\"\\\\';';
"
`;
